### DISCUSSION TITLE
Kong Admin API error from POD but accessible from browser

### DISCUSSION THREAD

**Discussion by 4NE3Q** [Kong/kong#12622] [Help] [answered]
Hi everyone,

So I'm trying to deploy kong on my bare metal cluster, I'm quite new to this so please bear with me.

I'm using this guide to setup kong in my k8s cluster:
https://docs.konghq.com/gateway/latest/install/kubernetes/proxy/

I have done most of the part but when I reach the step to install kong manager (which gets installed) the pod goes into crashloopbackoff and I have checked the logs for the pod and it says:
2024-02-25T18:42:22Z	info	setup	Retrying kong admin api client call after error	{"v": 0, "retries": "9/60", "error": "making HTTP request: Get \"http://localhost:8001/\": dial tcp [::1]:8001: connect: connection refused"}

However, I have already port forward the Admin API and I'm able to access the it from browser by going to localhost:8001 what could be wrong here if any one could help me deploy this and get around this error.

I'm able to access kong manager but obv it says no data to display as it is not connected to admin api as I see. I have also attached my values-cp.yaml with this and if someone want's to have a look at it ang guide me.
[values-cp.txt](https://github.com/Kong/kong/files/14397442/values-cp.txt)

Again, I would really appreciate any help in this, please note that this is a bare metal cluster and to get around kong-proxy as LoadBalancer I changed it to NodePort, not sure if that's related.

**Comment by hanshuebner** [comment]
Hi @4NE3Q,

from looking at your log output, it seems that Kong Manager uses `localhost` as the host for the admin API, which gets resolved to `[::1]` (IPv6 localhost).  Is it possible that Kong Gateway is configured to listen only on `127.0.0.1` (IPv4 localhost)?

We are aware that the current default for the `admin_listen` configuration variable is `127.0.0.1`, which is not optimal.  Before escalating this as a bug to be fixed, can you please confirm that things work for you if you set `admin_listen` to `[::1]:8001, 127.0.0.1:8001` (`KONG_ADMIN_LISTEN="[::1]:8001, 127.0.0.1:8001"`).

**Comment by 4NE3Q** [comment, marked as answer]
Fixed it myself. YAML file attaached.

[values-cp.txt](https://github.com/Kong/kong/files/14511524/values-cp.txt)
